driver-core/src/main/com/mongodb/client/model/Filters.java

```diff
@@ -1140,10 +1141,14 @@ public String toString() {
             }
         }
 
+        private boolean isDBRef(final BsonValue value) {
+            return value.asDocument().keySet().containsAll(DBREF_KEYS);
```

Comment:
This is a pretty naive check for a DBRef but as its a convention it should be enough.

Reply:
Ack.  I'm not sure what we're going to do once we start allowing $-prefixed fields in documents.